### DISCUSSION TITLE
How to enable Code profiling on a Lambda function?

### DISCUSSION THREAD

**Discussion by cytopia** [serverless/serverless#11074] [Q&A]
How would I enable "Code Profiling" on a Lambda on AWS (as available via GUI)?

![Screenshot 2022-05-13 16-51-45  selection](https://user-images.githubusercontent.com/12533999/168315624-75a94cb7-166e-4dd7-a95b-f8550da71aca.png)

**Comment by pgrzesik** [comment]
Hello @cytopia - this is currently not supported by the Framework - I'm not familiar with it enough to say if it's possible to enable it by using raw CloudFormation resources, but it seems like it might be possible via https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codeguruprofiler-profilinggroup.html

**Reply by cytopia** [reply]
> this is currently not supported by the Framework

Hi @pgrzesik appreciate the reply! So as it is currently not supported by the frameworks, do you have an outlook on when this will be integrated into serverless? For me it is specifically cruicial to make use of [AWS DevOps Guru](https://aws.amazon.com/devops-guru/) and have the Lambda applications data enhanced from [AWS Code Guru](https://aws.amazon.com/codeguru/) via its [Lambda profiling](https://docs.aws.amazon.com/lambda/latest/dg/monitoring-code-profiler.html) functionality.

**Reply by pgrzesik** [reply]
As long as it's supported by CloudFormation, you can still integrate with it via `resources` section by providing raw CloudFormation configuration that enables it for your functions. If it's not supported via CloudFormation, then we would have to wait for AWS to provide CloudFormation support for it first.

As for supporting it "natively" in the Framework, this is honestly the first time someone mentions it so we don't even have an issue that tracks it. 

I personally don't know much about this feature, so it would be great to provide an overview of the feature, e.g. if you'd like to submit an issue about it.

**Comment by cytopia** [comment]
@pgrzesik looks like there's a manual way available to programmatically enable this: https://docs.aws.amazon.com/codeguru/latest/profiler-ug/lambda-simple.html

It requires one to add:
1. additional environment variables
2. an additional lambda layer

I am going to try this out and will update in here if it works as expected
